rest/context.go

```diff
@@ -325,6 +351,60 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) buildURL(fullPath bool, resourceName string,
+	methodName string, pairs ...string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("unable to build URL for resource name %q: no request available.",
+			resourceName)
+	}
+
+	routeName := resourceName + ":" + methodName
+	route := ctx.router.Get(routeName).Host(r.Host)
+
+	var builder func(pairs ...string) (*url.URL, error)
```

Comment:
Theoretically you could declare a type alias such as

```go
type pairs map[string]string
```

and then use it like this:

```go
var p = pairs{"option1": "setting1", "option2": "setting2"}
```

But it's really @tylertreat-wf's call whether that makes sense for this library.